src/main/java/lesson7/FileNavigator.java

```diff
@@ -0,0 +1,86 @@
+package lesson7;
+
+import java.util.*;
+
+public class FileNavigator {
+
+    private final Map<String, List<FileData>> files = new HashMap<>();
+
+
+//2. Реалізувати метод add у класі FileNavigator. Цей метод додає файл за вказаним шляхом.
+// Якщо шлях вже існує, новий файл необхідно додати до списку, вже пов'язаному з відповідним шляхом.
+// ВАЖЛИВО: Шлях – унікальне значення і не повинно повторюватися.
+
+    public boolean add(String path, FileData file) {
+
+        if(files.containsKey(path)) {
+            List<FileData> existingFile = files.get(path);
+            List<FileData> newFile = new ArrayList<>(existingFile);
+            newFile.add(file);
+            files.put(path, newFile);
+        } else {
+            files.put(path, List.of(file));
+            return true;
+        }
+        return false;
+    }
+
+//   3. Реалізувати метод find у класі FileNavigator. Метод повертає список файлів,
+//   пов'язаних з шляхом переданим як параметр.
+
+    public List<FileData> find(String path) {
+        return files.get(path);
+    }
+
+//    4. Реалізувати метод filterBySize у класі FileNavigator. Метод повертає список файлів,
+//    розмір (в байтах) яких не перевищує значення, передане як параметр.
+
+    public List<FileData> filterBySize (int maxSize) {
+
+        List<FileData> filterFile = new ArrayList<>();
+
+        for (List<FileData> file : files.values()) {
+            for (FileData fileData : file) {
+                if (fileData.getSize() <= maxSize) {
+                    filterFile.add(fileData);
+                }
+            }
+        }
+        return filterFile;
+    }
+
+//   5. Реалізувати метод remove у класі FileNavigator. Метод видаляє шлях і пов'язані з ним файли,
+//   виходячи з значення шляху, переданого як параметр.
+
+    public boolean remove(String path) {
+        files.remove(path);
+        return true;
+    }
+
+//    6. * Реалізувати метод sortBySize у класі FileNavigator. Метод сортує всі наявні файли за розміром
+//    (за зростанням), потім повертає список відсортованих файлів.
+
+    public List<FileData> sortBySize() {
+
+        List<FileData> allFiles = new ArrayList<>();
+
+        for(List<FileData> fileData : files.values()) {
+            allFiles.add((FileData) fileData);
```

Comment:
я думаю це не працює, оскільки fileData в цей момент це List<FileData> а ти його кастиш до FileData
мабуть не вистачає теста на це.

Reply:
```suggestion
            allFiles.addAll(fileData);
```
має вирішити проблему